platform/admin/ui/src/main/webapp/js/models/Applications.js

```diff
@@ -132,7 +132,7 @@ define([
             changeObj.displayName = this.get('name');
             if (this.get('version') === '0.0.0') {
                 var matches = this.get('name').match(versionRegex);
-                if (matches.length === 3) {
+                if (matches !== null && matches.length === 3) {
```

Comment:
I think you can also remove the length check. If the match isn't null, then all the groups in the regex were found, so it will always be of length three.